src/schema/schema_drop.c

```diff
@@ -261,6 +261,12 @@ __drop_tiered(WT_SESSION_IMPL *session, const char *uri, bool force, const char
         __wt_free(session, name);
     }
 
+    /*
+     * If everything is successful to this point, remove any tiered work associated with this tiered
+     * handle before closing it.
+     */
+    __wt_tiered_remove_work(session, tiered);
```

Comment:
Sorry, I don't fully understand how the tiered server work queue works. What if the code fails after the line above? All the changes will be unrolled in `__meta_track_unroll`, but changes to the work queue will not be unrolled. Do you think it's OK?
I have the same concern about adding work units to the queue. What if we add an entry to the queue and then the code fails, do we have a cleanup code that removes elements from the work queue?

Reply:
I will give this some thought. I think `meta_track` is not the right place for changes to the work queue but also that you're on the right track with considering how error handling is done. We may need some kind of `tiered_track` or other tracking mechanism similar to metadata.

Reply:
Logically, these changes to the work queue are the result of successfully dropping the tiered table.  So we shouldn't queue them until we know the drop has completed -- i.e., after or during the call to `wt_meta_track_off`. Queuing them earlier might mean that some of the work items get erroneously applied in a case where we unroll the metadata changes, effectively canceling the drop.

I'm not sure that in this case (or similar ones) it would help to have a separate way to track and unroll work queue changes, since the triggering event is a metadata change.

Do we need anything fancy for error handling outside of metadata operations?  E.g., in `tiered_restart_work` and `tier_do_operation`.

Reply:
The problem with doing it as late as `wt_meta_track_off` is that then the dhandle is already closed and freed. That still leaves the window open for the internal thread using a bad tiered structure. That's why this call is added immediately before the dhandle release and after all metadata changes are done.

But maybe if those get an error, there can be a special form of `tiered_restart_work` just for this one table.

Reply:
And `tiered_restart_work` already is coded for just one table.

Reply:
I've committed code and updated the ticket with the path I chose and a 2nd path I considered.